### PR TITLE
add user to group IDs bulk action

### PR DESCRIPTION
**Pull request by jdf400-rapid7**
### Description
Add user to groups by ID bulk actuon
Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [ ] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [ ] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [ ] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR.
4. Add UI screenshot of the workflow used for testing
5. Add UI screenshot of the job output used for testing
6. Add UI screenshot of the artifact (See rules in UI Checklist) used for testing

### CODE REVIEW

**Comment by jdf400-rapid7** [comment]
⇒  icon-plugin run -R tests/add_user_to_groups_by_id.json -A
INFO[0000] Running command:  docker run --rm -i rapid7/azure_ad_admin:2.1.0 run < tests/add_user_to_groups_by_id.json 
{"body": {"log": "Connect: Connecting...\nUpdating auth token...\nGetting token from: https://login.microsoftonline.com/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication request status: 200\nAuthentication Token: ****************jN1eQ\nrapid7/Azure AD Admin:2.1.0. Step name: add_user_to_groups_by_id\nGetting user info: jschipp@komanddev.onmicrosoft.com\n", "status": "ok", "meta": {}, "output": {"success": true}}, "version": "v1", "type": "action_event"}

## Assessment
### Run

<details>

```
{
  "body": {
    "log": "Connect: Connecting...\nUpdating auth token...\nGetting token from: https://login.microsoftonline.com/5c824599-dc8c-4d31-96fb-3b886d4f8f10/oauth2/token\nAuthentication request status: 200\nAuthentication Token: ****************jN1eQ\nrapid7/Azure AD Admin:2.1.0. Step name: add_user_to_groups_by_id\nGetting user info: jschipp@komanddev.onmicrosoft.com\n",
    "meta": {},
    "output": {
      "success": true
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/azure_ad_admin:2.1.0 run < tests/add_user_to_groups_by_id.json
</summary>
</details>

### UI

Screenshots of the plugin being used in Komand for validating use go below this line.

#### Workflow Builder


#### Job


#### Artifact

File: azure_ad_admin/help.md [hunk 1]
```diff
@@ -35,6 +35,33 @@ The connection configuration accepts the following parameters:
...
+
+##### Output
+
+|Name|Type|Required|Description|
+|----|----|--------|-----------|
+|success|boolean|False|Was operation successful|
+
+Example output:
+
+```
```
Comment (jschipp-r7) [inline azure_ad_admin/help.md:62]: Needs example output for the new action

File: azure_ad_admin/help.md [hunk 2]
```diff
@@ -35,6 +35,33 @@ The connection configuration accepts the following parameters:
...
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|Example|
+|----|----|-------|--------|-----------|----|-------|
+|group_id|[]string|None|True|IDs of Groups to Add User to|None|\["b4d41d4-eb13-4a33-99b5-7d7290df22e9"\]|
+|user_id|string|None|True|User ID e.g. user@example.com|None|user@example.com|
+
+Example input:
+
+```
```
Comment (jschipp-r7) [inline azure_ad_admin/help.md:51]: Needs example input for the new action

File: azure_ad_admin/plugin.spec.yaml [hunk 3]
```diff
@@ -451,6 +451,28 @@ actions:
...
+        description: User ID e.g. user@example.com
+        type: string
+        required: true
+        example: user@example.com
+      group_id:
+        title: Group IDs
+        description: IDs of Groups to Add User to
+        type: "[]string"
+        required: true
+        example: \["b4d41d4-eb13-4a33-99b5-7d7290df22e9"\]
```
Comment (jschipp-r7) [inline azure_ad_admin/plugin.spec.yaml:469]:
```suggestion
        example: ["b4d41d4-eb13-4a33-99b5-7d7290df22e9"]
```

File: azure_ad_admin/plugin.spec.yaml [hunk 4]
```diff
@@ -451,6 +451,28 @@ actions:
         description: Was operation successful
         type: boolean
         required: false
+  add_user_to_groups_by_id:
+    title: Add User to Groups by ID
+    description: Add a user to a groups by ID
```
Comment (jschipp-r7) [inline azure_ad_admin/plugin.spec.yaml:456]:
```suggestion
    description: Add a user to a set of groups by group ID
```

**Comment by jschipp-r7** [comment]
LGTM 👍

**Pull request merged**